packages/edit-site/src/components/page-pages/index.js

```diff
@@ -39,19 +44,74 @@ const defaultConfigPerViewType = {
 	},
 };
 
-export default function PagePages() {
+const PATH_TO_TYPE = {
+	'/pages': 'page',
+};
+
+function useView() {
 	const {
-		params: { path, activeView = 'all' },
+		params: { path, activeView = 'all', isCustom = 'false' },
 	} = useLocation();
-	const initialView = DEFAULT_VIEWS.find(
-		( { slug } ) => slug === activeView
-	).view;
-	const [ view, setView ] = useState( initialView );
+	const type = PATH_TO_TYPE[ path ];
+	const selectedDefaultView =
+		isCustom === 'false' &&
+		DEFAULT_VIEWS[ type ].find( ( { slug } ) => slug === activeView )?.view;
+	const [ view, setView ] = useState( selectedDefaultView );
+
+	const customViews = useCustomDataViews( type );
 	useEffect( () => {
-		setView(
-			DEFAULT_VIEWS.find( ( { slug } ) => slug === activeView ).view
+		if ( selectedDefaultView ) {
+			setView( selectedDefaultView );
+		}
+	}, [ selectedDefaultView ] );
+	const selectedCustomViewRecord =
+		isCustom === 'true' &&
+		customViews.find( ( { slug } ) => slug === activeView );
+	const editedViewRecord = useSelect(
+		( select ) => {
+			if ( ! selectedCustomViewRecord ) {
+				return;
+			}
+			const { getEditedEntityRecord } = select( coreStore );
+			const dataviewRecord = getEditedEntityRecord(
+				'postType',
+				'wp_dataviews',
+				selectedCustomViewRecord.id
 		);
-	}, [ path, activeView ] );
+			return dataviewRecord;
+		},
+		[ selectedCustomViewRecord ]
+	);
+	const { editEntityRecord } = useDispatch( coreStore );
+
+	const [ customView, customSetView ] = useMemo( () => {
+		return [
+			editedViewRecord?.content &&
+				JSON.parse( editedViewRecord?.content ),
+			function ( viewToSet ) {
+				editEntityRecord(
+					'postType',
+					'wp_dataviews',
+					editedViewRecord?.id,
+					{
+						content: JSON.stringify( viewToSet ),
+					}
+				);
+			},
+		];
+	}, [ editEntityRecord, editedViewRecord?.content, editedViewRecord?.id ] );
```

Comment:
I think it's not great that the onChange depend on the value itself. Ideally the customView and setCustomView should be memoized separately.

Reply:
Updated to memoize separately.